src/selectors/entities/channels.js

```diff
@@ -306,20 +306,33 @@ export const getMembersInCurrentChannel = createSelector(
 export const getUnreads = createSelector(
     getAllChannels,
     getMyChannelMemberships,
-    (channels, myMembers) => {
+    getUsers,
+    getCurrentUserId,
+    (channels, myMembers, users, currentUserId) => {
         let messageCount = 0;
         let mentionCount = 0;
         Object.keys(myMembers).forEach((channelId) => {
             const channel = channels[channelId];
             const m = myMembers[channelId];
+
             if (channel && m) {
+                let otherUserId;
                 if (channel.type === 'D') {
-                    mentionCount += channel.total_msg_count - m.msg_count;
+                    otherUserId = getUserIdFromChannelName(currentUserId, channel.name);
+                    if (users[otherUserId]) {
+                        mentionCount += channel.total_msg_count - m.msg_count;
+                    }
                 } else if (m.mention_count > 0) {
                     mentionCount += m.mention_count;
                 }
                 if (m.notify_props && m.notify_props.mark_unread !== 'mention' && channel.total_msg_count - m.msg_count > 0) {
-                    messageCount += 1;
+                    if (channel.type === 'D') {
+                        if (users[otherUserId]) {
```

Comment:
same here

Reply:
ok I can see what you mean..

2 things:
1. needs rebase
2. should we use the release-5.2 branch for this?